File: get-started/azure-auth/php/GetUser.php
<!-- Copyright (c) Microsoft Corporation.
     Licensed under the MIT License.-->

<!-- <ProgramSnippet> -->
<?php

use GetUser\Client\GraphApiClient;
use Microsoft\Kiota\Abstractions\ApiException;
use Microsoft\Kiota\Authentication\Oauth\AuthorizationCodeContext;
use Microsoft\Kiota\Authentication\PhpLeagueAuthenticationProvider;
use Microsoft\Kiota\Http\GuzzleRequestAdapter;

require __DIR__.'/vendor/autoload.php';

try {
    $clientId = 'clientId';
    $clientSecret = 'secret';
    $authorizationCode = 'authCode';

    $tenantId = 'common';
    $redirectUri = 'http://localhost';

    // The auth provider will only authorize requests to
    // the allowed hosts, in this case Microsoft Graph
    $allowedHosts = ['graph.microsoft.com'];
    $scopes = ['User.Read'];

    $tokenRequestContext = new AuthorizationCodeContext(
        $tenantId,
        $clientId,
        $clientSecret,
        $authorizationCode,
        $redirectUri
    );

    $authProvider = new PhpLeagueAuthenticationProvider($tokenRequestContext, $scopes, $allowedHosts);
    $requestAdapter = new GuzzleRequestAdapter($authProvider);
    $client = new GraphApiClient($requestAdapter);

    $me = $client->me()->get()->wait();
    echo "Hello {$me->getDisplayName()}, your ID is {$me->getId()}";

} catch (ApiException $ex) {
    echo $ex->getMessage();
}
?>
<!-- </ProgramSnippet> -->

File: get-started/azure-auth/ruby/get_user.rb
# Copyright (c) Microsoft Corporation.
# Licensed under the MIT License.

# <ProgramSnippet>
# frozen_string_literal: true
require 'microsoft_kiota_serialization_json'
require 'microsoft_kiota_abstractions'
require 'microsoft_kiota_authentication_oauth'
require 'microsoft_kiota_faraday'
require_relative './client/get_user_api_client'

client_id = 'CLIENT_ID'
client_secret = 'CLIENT_SECRET'
auth_code = 'AUTH_CODE'

tenant_id = 'common'
redirect_uri = 'http://localhost'

# The auth provider will only authorize requests to
# the allowed hosts, in this case Microsoft Graph
allowed_hosts = ['graph.microsoft.com']
graph_scopes = ['User.Read']
token_request_context = MicrosoftKiotaAuthenticationOAuth::AuthorizationCodeContext
  .new(tenant_id, client_id, client_secret, redirect_uri, auth_code)

auth_provider = MicrosoftKiotaAuthenticationOAuth::OAuthAuthenticationProvider
  .new(token_request_context, allowed_hosts, graph_scopes)

request_adapter = MicrosoftKiotaFaraday::FaradayRequestAdapter
  .new(auth_provider,
       MicrosoftKiotaSerializationJson::JsonParseNodeFactory.new,
       MicrosoftKiotaSerializationJson::JsonSerializationWriterFactory.new)

client = GetUser::GetUserApiClient.new(request_adapter)

me = client.me().get().resume

puts "Hi! My name is #{me.display_name}, and my ID is #{me.id}."
# </ProgramSnippet>
